docs/remove_controller.md

```diff
@@ -41,6 +41,61 @@ k0s reset
 reboot
 ```
 
+### Declarative Etcd member management
+
+Starting from version 1.30, k0s also supports a declarative way to remove an  
+etcd member. Since in k0s the etcd cluster is set up so that the etcd API is  
+**NOT** exposed outside the nodes, it makes it difficult for external automation  
+like Cluster API, Terraform, etc. to handle controller node replacements.
+
+Each controller manages their own `EtcdMember` object.
+
+```shell
+k0s kubectl get etcdmember
+NAME          PEER ADDRESS   MEMBER ID           JOINED   RECONCILE STATUS
+controller0   172.17.0.2     b8e14bda2255bc24    True     
+controller1   172.17.0.3     cb242476916c8a58    True     
+controller2   172.17.0.4     9c90504b1bc867bb    True 
+```
+
+By marking an `EtcdMember` object to leave the etcd cluster, k0s will handle the  
+interaction with etcd. For example, in a 3 controller HA setup, you can  
+remove a member by flagging it to leave:
+
+```console
+$ kubectl patch etcdmember controller2 -p '{"spec":{"leave":true}}' --type merge
+etcdmember.etcd.k0sproject.io/controller2 patched
+```
+
+The join/leave status is tracked in the object's conditions. This allows you to  
+wait for the leave to actually happen:
+
+```console
+$ kubectl wait etcdmember controller2 --for condition=Joined=False
+etcdmember.etcd.k0sproject.io/controller2 condition met
+```
+
+You'll see the node left etcd cluster:
+
+```console
+$ k0s kc get etcdmember
+NAME          PEER ADDRESS   MEMBER ID           JOINED   RECONCILE STATUS
+controller0   172.17.0.2     b8e14bda2255bc24    True     
```

Comment:
I wonder if k0s should also set the reconcile status to Success when joined is true.